src/audio/module_adapter/module_adapter.c

```diff
@@ -693,95 +693,172 @@ module_single_source_setup(struct comp_dev *dev,
-	} else {
-		list_for_item(blist, &mod->sink_buffer_list) {
-			sink = container_of(blist, struct comp_buffer, sink_list);
+	list_for_item(blist, &mod->sink_buffer_list) {
+		sink = container_of(blist, struct comp_buffer, sink_list);
 
-			sink_c = buffer_acquire(sink);
-			min_free_frames = MIN(min_free_frames,
-					      audio_stream_get_free_frames(&sink_c->stream));
-			buffer_release(sink_c);
-		}
+		sink_c = buffer_acquire(sink);
+		min_free_frames = MIN(min_free_frames,
+				      audio_stream_get_free_frames(&sink_c->stream));
+		buffer_release(sink_c);
+	}
 
-		/* copy source samples into input buffer */
-		list_for_item(blist, &dev->bsource_list) {
-			struct comp_buffer __sparse_cache *src_c;
-			uint32_t bytes_to_process;
-			int frames, source_frame_bytes;
+	/* copy source samples into input buffer */
+	list_for_item(blist, &dev->bsource_list) {
+		struct comp_buffer __sparse_cache *src_c;
+		uint32_t bytes_to_process;
+		int frames, source_frame_bytes;
 
-			source = container_of(blist, struct comp_buffer, sink_list);
-			src_c = buffer_acquire(source);
+		source = container_of(blist, struct comp_buffer, sink_list);
+		src_c = buffer_acquire(source);
 
-			/* check if the source dev is in the same state as the dev */
-			if (!src_c->source || src_c->source->state != dev->state) {
-				buffer_release(src_c);
-				continue;
-			}
+		/* check if the source dev is in the same state as the dev */
+		if (!src_c->source || src_c->source->state != dev->state) {
+			buffer_release(src_c);
+			continue;
+		}
 
-			frames = MIN(min_free_frames,
-				     audio_stream_get_avail_frames(&src_c->stream));
-			source_frame_bytes = audio_stream_frame_bytes(&src_c->stream);
+		frames = MIN(min_free_frames,
+			     audio_stream_get_avail_frames(&src_c->stream));
+		source_frame_bytes = audio_stream_frame_bytes(&src_c->stream);
 
-			bytes_to_process = MIN(frames * source_frame_bytes, md->mpd.in_buff_size);
+		bytes_to_process = MIN(frames * source_frame_bytes, md->mpd.in_buff_size);
 
-			buffer_stream_invalidate(src_c, bytes_to_process);
-			mod->input_buffers[i].size = bytes_to_process;
-			mod->input_buffers[i].consumed = 0;
+		buffer_stream_invalidate(src_c, bytes_to_process);
+		mod->input_buffers[i].size = bytes_to_process;
+		mod->input_buffers[i].consumed = 0;
 
-			ca_copy_from_source_to_module(&src_c->stream, mod->input_buffers[i].data,
-						      md->mpd.in_buff_size, bytes_to_process);
-			buffer_release(src_c);
+		ca_copy_from_source_to_module(&src_c->stream, mod->input_buffers[i].data,
+					      md->mpd.in_buff_size, bytes_to_process);
+		buffer_release(src_c);
 
-			i++;
-		}
-		num_input_buffers = mod->num_input_buffers;
-		num_output_buffers = mod->num_output_buffers;
+		i++;
 	}
 
-	ret = module_process(mod, mod->input_buffers, num_input_buffers,
-			     mod->output_buffers, num_output_buffers);
+	ret = module_process(mod, mod->input_buffers, mod->num_input_buffers,
+			     mod->output_buffers, mod->num_output_buffers);
```

Comment:
I think there's a change in behaviour here: before this patch if the loop always took the `continue` path because all components were in wrong states, then `num_input_buffers` and `num_output_buffers` were 0 here, now that's different. Is that intended?

Reply:
@lyakh but that was only for the simple_copy case right?